src/Controller.js

```diff
@@ -0,0 +1,123 @@
+import FunctionTree from 'function-tree'
+import Module from './Module'
+import DefaultModel from './DefaultModel'
+import {ensurePath, isDeveloping, throwError} from './utils'
+import VerifyInputProvider from './providers/VerifyInput'
+import ContextProvider from 'function-tree/providers/Context'
+import StateProvider from './providers/State'
+import DebuggerProvider from './providers/Debugger'
+import ControllerProvider from './providers/Controller'
+import {EventEmitter} from 'events'
+import {dependencyStore as computedDependencyStore} from './Computed'
+
+/*
+  The controller is where everything is attached. The devtools
+  and router is attached directly. Also a top level module is created.
+  The controller creates the function tree that will run all signals,
+  based on top level providers and providers defined in modules
+*/
+class Controller extends EventEmitter {
+  constructor({state = {}, routes = {}, signals = {}, providers = [], modules = {}, router, devtools}) {
+    super()
+    this.flush = this.flush.bind(this)
+    this.devtools = devtools
+    this.model = new DefaultModel({})
```

Comment:
I'm assuming this is the mutable model? We ran into one issue using the mutable model, and maybe we can add an option which freezes it, or copies it once you retrieve it from the store. 

```
function({state}) {
 var user = state.get("user");
 user.name = "Modified";
}
```
and because user is a reference to an object it actually changes it in the store. Soo easy to make this mistake, so we have turn on the immutability one now so that if anyone does this accidentally it throws an error.

Thoughts?